src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfig/EditorConfigStorageLocation.cs

```diff
@@ -12,9 +12,6 @@ internal static class EditorConfigStorageLocation
         public static EditorConfigStorageLocation<bool> ForBoolOption(string keyName)
```

Comment:
there seem to be too much logic duplication here - all the parsing (except for the notification part) should be coming from the editorconfigdata object.